### PR TITLE
[WEEX-86] Update documentation for contribution.

### PR DESCRIPTION
**Pull request by cxfeng1**
Separate contributing to several ways: reporting bugs, answering questions, joining in discussions, contributing documentation and contributing code.
This will make it easier for developers who just want to report bugs,  and it will also make the contributing process progressive.
Two additional documents are also added: Bug Report Guidelines and Development Process,  which provide more detail information about reporting bugs and contributing code.

issue: 86

### CODE REVIEW

File: source/bug-report-guidelines.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+---
+title: Bug Report Guidelines
+type: community
+has_chapter_content: false
+version: 2.1
+---
+
+# Bug Report Guidelines
+
+This document describes how to write a good Weex bug report. Good bug reports help developers decide an appropriate priority and severity for a bug, and increase the chance that a bug will be fixed quickly. The more specific information you can provide, the better.
```
Comment (jondong) [inline source/bug-report-guidelines.md]: to classify the priority and severity of a bug properly, which helps the bug get fixed as soon as possible. The more specific information you provide, the better your bug gets understood.

File: source/bug-report-guidelines.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+## Bug Report General Principles
+
+- Avoid duplicates: Search before you file!
```
Comment (jondong) [inline source/bug-report-guidelines.md]: before you file a new one.

File: source/bug-report-guidelines.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+- Always test the latest available version.
```
Comment (jondong) [inline source/bug-report-guidelines.md]: before submitting a bug

File: source/bug-report-guidelines.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+- One bug per report.
+- State useful facts, not opinions or complaints.
+
+## How to Write a Good Bug Report
+
+A good bug report should include the following information:
+
+### Summary
```
Comment (jondong) [inline source/bug-report-guidelines.md]: is it better to use title here? not quite sure

File: source/bug-report-guidelines.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+The goal of summary is to make the report searchable and uniquely identifiable.
+
+A bad example: `List Crash`
+
+A good Example: `List Crashes when deleting a header`
+
+### Environment
+
+**Weex Version**: Please tell the version of WeexSDK or Weex Playground or weex-toolkit you were using when the bug occurred
```
Comment (jondong) [inline source/bug-report-guidelines.md]: please identify...

File: source/bug-report-guidelines.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+**Device environment**: Please tell the device model, platform and OS version that your bug occurred on. e.g. , iPhone 6, iOS 10.3.
```
Comment (jondong) [inline source/bug-report-guidelines.md]: identify.
remove "that your bug occurred on"

File: source/bug-report-guidelines.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+
+### Overview/Description
+
+The overview or description of a bug report is to explain the bug to the developer, including:
```
Comment (jondong) [inline source/bug-report-guidelines.md]: is to explain the bug in detail.
remove "to the developer"

File: source/bug-report-guidelines.md [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+
+- Justifications of why this is a bug
+- [dotwe](http://dotwe.org/vue/) demo that can reproduce the bug
+- Screenshots for visual bugs
+- Stack traces for crash bugs
+- Console logs or error messages which are relevant in the bug
+
+### Steps to Reproduce
+
+The goal of reproducible steps is to teach developer to recreate the bug on his own system, such as:
```
Comment (jondong) [inline source/bug-report-guidelines.md]: The aim to provide the reproducible steps is to enable developers to reproduce the bug in their own environment. Here's an example:

File: source/bug-report-guidelines.md [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+
+*Step 1: Load the demo using Weex Playground*
+
+*Step 2: Scroll to the bottom of the list*
+
+*Step 3: Click the red button to delete a header*
+
+### Test Results
+
+The test results, including *Expected Result* and *Actual Result*, will show the developer what's wrong. *Expected Result* describes what should have happened, and *Actual Result* describes what actually happened.
```
Comment (jondong) [inline source/bug-report-guidelines.md]: will tell developers what's wrong.

File: source/bug-report-guidelines.md [hunk 10]
```diff
@@ -56,7 +56,7 @@ The goal of reproducible steps is to teach developer to recreate the bug on his
 
 ### Test Results
 
-The test results, including *Expected Result* and *Actual Result*, will show the developer what's wrong. *Expected Result* describes what should have happened, and *Actual Result* describes what actually happened.
+The test results, including *Expected Result* and *Actual Result*, will tell developers what's wrong. *Expected Result* describes what should have happened, and *Actual Result* describes what actually happened.
```
Comment (jondong) [inline source/bug-report-guidelines.md]: what should happen, & what actually happens.
I think we are describing the common fact, it is better to use the present tense.

File: source/bug-report-guidelines.md [hunk 11]
```diff
@@ -7,12 +7,12 @@ version: 2.1
 
 # Bug Report Guidelines
 
-This document describes how to write a good Weex bug report. Good bug reports help developers decide an appropriate priority and severity for a bug, and increase the chance that a bug will be fixed quickly. The more specific information you can provide, the better.
+This document describes how to write a good Weex bug report. Good bug reports help developers classify the priority and severity of a bug properly, which helps the bug get fixed as soon as possible. The more specific information you provide, the better your bug gets understood.
```
Comment (jondong) [inline source/bug-report-guidelines.md]: to classify

**Comment by Hanks10100** [comment]
This PR is already merged into master.

**Pull request closed without merging**